article/_posts/2020-09-23-G-SGD.md

```diff
@@ -0,0 +1,139 @@
+---
+layout: review
+title: "G-SGD: Optimizing ReLU neural networks in its positively scale-invariant space"
+tags: optimization 
+author: "Marco Armenta"
+cite:
+    authors: "Qi Meng, Shuxin Zheng, Huishuai Zhang, Wei Chen, Qiwei Ye, Zhi-Ming Ma, Nenghai Yu, Tie-Yan Liu"
+    title:   "G-SGD: Optimizing ReLU neural networks in its positively scale-invariant space"
+    venue:   "ICLR 2019"
+pdf: "https://openreview.net/pdf?id=SyxfEn09Y7"
+---
+
+# Highlights
+- ReLU neural networks are optimized in a space that is invariant under positive rescaling. This space is also sufficient to represent ReLU networks.
+- This is achieved by exploiting dependencies between the values of all the paths from the input layer to the output layer.
+- This method only works for MLPs and CNNs with the same number of neurons per hidden layer.
+
+# Introduction
+
+Let $$N_w(x): \mathcal{X} \to \mathcal{Y}$$ denote a L-layer multi-layer perceptron with weights $$w \in \mathcal{W} \subset \mathbb{R}^m$$, input space $$\mathcal{X} \subset \mathcal{R}^d$$ and output space $$\mathcal{Y} \subset \mathcal{R}^K$$. In the $$l$$-th layer, there are $$h_l$$ neurons. We denote the $$i_l$$-th node by $$O_{i_l}^l$$ and its value on a forward pass by $$o_{i_l}^l$$. The weight matrix between layers $$l-1$$ and $$l$$ is denoted $$w^l$$, and the weight connecting neurons $$O_{i_l-1}^{l-1}$$ and $$O_{i_l}^l$$ is denoted $$w^l(i_{l-1},i_{l})$$. The path from the input node $$O_{i_0}^0$$ to the output node $$O_{i_L}^L$$ passing through the nodes $$O_{i_0}^0,...,O_{i_L}^L$$ is denoted by $$(i_0,...,i_L)$$.
+The network structure can be regarded as a directed graph $$(\mathcal{O},E)$$, with vertices $$\mathcal{O}=\{ O_1,...,O_{H+d+K} \}$$ and edges $$E$$, where H is the number of hidden neurons and $$|E|=m$$.
+
+A path is represented by a vector $$p=(p_1,...,p_m)^T$$ and if edge e is contained in path p, then $$p_e=1$$ and otherwise $$p_e=0$$. These vectors have L entries that are non-zero. The value of path p is defined as $$v_p(w)= \prod_{i=1}^m w_i^{p_i}$$. The set of all such paths is denoted $$\mathcal{P}$$. Finally, the activation status of path $$p$$ is $$a_p(x;w)=\prod_{j:p_{e_{ij}}=1} \mathbb{I}\big( o_j(x;w) > 0 \big)$$.
```

Comment:
edge $$e$$ and path $$p$$ [...] $$L$$ entries

Reply:
Done!